src/hidimstat/dcrt.py

```diff
@@ -11,265 +11,355 @@
-    Sigma_X=None,
+    sigma_X=None,
     use_cv=False,
     cv=5,
     n_alphas=20,
     alpha=None,
+    tol=1e-6,
     max_iter=1000,
     refit=False,
-    loss="least_square",
     screening=True,
     screening_threshold=1e-1,
-    scaled_statistics=False,
     statistic="residual",
     centered=True,
-    fdr_control="bhq",
     n_jobs=1,
-    verbose=False,
     joblib_verbose=0,
+    fit_y=False,
     ntree=100,
     problem_type="regression",
     random_state=2022,
 ):
     """
-    This function implements the Conditional Randomization Test of
-    :footcite:t:`candesPanningGoldModelX2017` accelerated with the distillation
-    process `dcrt_zero` in the work by :footcite:t:`liuFastPowerfulConditional2021`.
+    Implements distilled conditional randomization test (dCRT) without interactions.
+
+    A faster version of the Conditional Randomization Test :cite:`candes2018panning` using the distillation
+    process from :cite:`liu2022fast`. Based on original implementation at:
+    https://github.com/moleibobliu/Distillation-CRT/
 
     Parameters
     ----------
-    X : {array-like, sparse matrix} of shape (n_samples, n_features)
-        The input samples.
+    X : array-like of shape (n_samples, n_features)
+        Training data
     y : array-like of shape (n_samples,)
-        The target values (class labels in classification, real numbers in
-        regression).
-    fdr : float, default=0.1
-        The desired controlled FDR level.
-    estimated_coef : array-like of shape (n_features,)
-        The array of the corresponding coefficients for the features.
-    Sigma_X : {array-like, sparse matrix} of shape (n_features, n_features)
-        The covariance matrix of X.
+        Target values
+    estimated_coef : array-like of shape (n_features,), optional
+        Pre-computed feature coefficients
+    sigma_X : array-like of shape (n_features, n_features), optional
+        Covariance matrix of X
     use_cv : bool, default=False
-        Whether to apply cross-validation for the distillation with Lasso.
-    cv : int, cross-validation generator or iterable, default=None
-        Determines the cross-validation splitting strategy. Possible inputs for
-        cv are:
-        - None, to use the default 5-fold cross-validation,
-        - int, to specify the number of folds.
-        - CV splitter,
-        - An iterable yielding (train, test) splits as arrays of indices.
+        Whether to use cross-validation for Lasso
+    cv : int, default=5
+        Number of cross-validation folds
     n_alphas : int, default=20
-        The number of alphas along the regularization path for
-        sklearn.linear_model.LassoCV().
-    alpha : float, default=None
-        Constant that multiplies the L1 term, controlling regularization
-        strength. alpha must be a non-negative float i.e. in [0, inf).
+        Number of alphas for Lasso path
+    alpha : float, optional
+        L1 regularization strength
```

Comment:
Similar to the sklearn API, I suggest that we add an `alphas` argument so that the regularization path is exposed to the user. alternatively `n_alphas` would lead to the precomputed grid.

Reply:
done